### PR TITLE
[Common/Util] function to get possible caps @open sesame 07/06 09:50

### PR DESCRIPTION
**Pull request by jaeyun-jung**
Fix nego error case, add new function to get possible caps from config.
When receives caps event, element should return all possible caps.

Signed-off-by: Jaeyun <jy1210.jung@samsung.com>

### CODE REVIEW

File: gst/nnstreamer/tensor_common.h [hunk 1]
```diff
@@ -174,6 +174,17 @@ gst_tensor_time_sync_buffer_from_collectpad (GstCollectPads * collect, tensor_ti
 extern GstCaps *
 gst_tensor_pad_caps_from_config (GstPad * pad, const GstTensorsConfig * config);
 
+/**
+ * @brief Get all possible caps from tensor config. Unlike gst_tensor_pad_caps_from_config(), this function does not check peer caps.
```
Comment (wooksong) [inline gst/nnstreamer/tensor_common.h:178]: This is a trivial one.
```suggestion
 * @brief Get all possible caps from tensors config. Unlike gst_tensor_pad_caps_from_config(), this function does not check peer caps.
```

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_common.h]: updated

**Review by gichan-jang — APPROVED** [review]
LGTM

File: gst/nnstreamer/tensor_filter/tensor_filter.c [hunk 2]
```diff
@@ -1047,6 +1004,11 @@ gst_tensor_filter_transform_caps (GstBaseTransform * trans,
   silent_debug_caps (caps, "from");
   silent_debug_caps (filter, "filter");
 
+  if (direction == GST_PAD_SINK)
```
Comment (wooksong) [inline gst/nnstreamer/tensor_filter/tensor_filter.c:1007]: If ```direction == GST_PAD_SINK```, then ```pad = GST_BASE_TRANSFORM_SRC_PAD (trans)```.
Is it right?

File: gst/nnstreamer/tensor_filter/tensor_filter.c [hunk 3]
```diff
@@ -1047,6 +1004,11 @@ gst_tensor_filter_transform_caps (GstBaseTransform * trans,
...
+    pad = GST_BASE_TRANSFORM_SRC_PAD (trans);
```
Comment (wooksong) [inline gst/nnstreamer/tensor_filter/tensor_filter.c:1008]: If ```direction == GST_PAD_SINK```, then ```pad = GST_BASE_TRANSFORM_SRC_PAD (trans)```.
Is it right?

Comment (jaeyun-jung) [reply gst/nnstreamer/tensor_filter/tensor_filter.c:1008]: yes, direction to sink then return src pad caps.

**Review by dongju-chae — APPROVED** [review]
LGTM

**Pull request merged**